cas-server-webapp-validation/src/main/java/org/jasig/cas/web/AbstractServiceValidateController.java

```diff
@@ -128,13 +151,38 @@ protected Credential getServiceCredentialsFromRequest(final WebApplicationServic
                 verifyRegisteredServiceProperties(registeredService, service);
                 return new HttpBasedServiceCredential(new URL(pgtUrl), registeredService);
             } catch (final Exception e) {
-                logger.error("Error constructing pgtUrl", e);
+                logger.error("Error constructing {}", CasProtocolConstants.PARAMETER_PROXY_CALLBACK_URL, e);
             }
         }
 
         return null;
     }
 
+
+    /**
+     * Validate authentication context pair.
+     *
+     * @param assertion the assertion
+     * @param request   the request
+     * @return the pair
+     */
+    protected Pair<Boolean, Optional<MultifactorAuthenticationProvider>> validateAuthenticationContext(
+            final Assertion assertion, final HttpServletRequest request) {
+        final String requestedContext = request.getParameter(this.authenticationContextParameter);
```

Comment:
well, I see we do load the `RegisteredService` below.

but the `!StringUtils.hasLength(requestedContext)` test will short-circuit if the client hasn't been updated to send the additional parameter to enforce MFA. This means client services would need to be updated and it couldn't be completely centrally configured without needing that update.

, but this still requires the end client to specify that it wants to enforce MFA, which means client updates